snap/snapcraft.yaml

```diff
@@ -74,14 +73,20 @@ parts:
     build-packages: [gcc, libffi-dev, libssl-dev, git, libaugeas-dev, python3-dev]
     build-environment:
       - SNAPCRAFT_PYTHON_VENV_ARGS: --upgrade
-      - PIP_NO_BUILD_ISOLATION: "no"
+      # Constraints are passed through the environment variable PIP_CONSTRAINTS instead of using the
+      # parts.[part_name].constraints available in snapcraft.yaml when Python plugin is used, in order
+      # to let these constraints be applied not only on the certbot package build, but also on any
+      # isolated build that pip could trigger to construct transitively dependant wheels.
+      - PIP_CONSTRAINT: $SNAPCRAFT_PART_SRC/snap-constraints.txt
     override-build: |
       python3 -m venv "${SNAPCRAFT_PART_INSTALL}"
-      "${SNAPCRAFT_PART_INSTALL}"/bin/python3 "${SNAPCRAFT_PART_SRC}/tools/pipstrap.py"
+      "${SNAPCRAFT_PART_INSTALL}/bin/python3" "${SNAPCRAFT_PART_SRC}/tools/pipstrap.py"
       snapcraftctl build
     override-pull: |
       snapcraftctl pull
-      python3 "${SNAPCRAFT_PART_SRC}/tools/strip_hashes.py" "${SNAPCRAFT_PART_SRC}/letsencrypt-auto-source/pieces/dependency-requirements.txt" | grep -v python-augeas > "${SNAPCRAFT_PART_SRC}/snap-constraints.txt"
+      python3 "${SNAPCRAFT_PART_SRC}/tools/strip_hashes.py" "${SNAPCRAFT_PART_SRC}/letsencrypt-auto-source/pieces/dependency-requirements.txt" | grep -v python-augeas >> "${SNAPCRAFT_PART_SRC}/snap-constraints.txt"
```

Comment:
I do not have a strong preference. I wanted to make a little clearer that the "final" result is the one from `merge_requirements`, since it is the only one overwritting the constraint file content.

Reply:
For what it's worth, the reason I prefer overwriting the file is:

1. It makes it clear reading this code that we don't expect (or care) if there are current contents there.
2. If someone has a `snap-constraints.txt` file in the root of their repo locally, it doesn't affect the build.